### ISSUE TITLE
Derived store gets out of sync with origin stores if some subscriber of origin store immediately changes its value

### ISSUE THREAD

**Issue by CherryDT** [PixievoltNo1/svelte-writable-derived#18]
Easiest example, using `propertyStore`:

**testStore.js**

```javascript
import { writable } from 'svelte/store'
import { propertyStore } from 'svelte-writable-derived'

export const outerStore = writable({ inner: '?' })
export const innerStore = propertyStore(outerStore, 'inner')
```

**App.svelte**

```html
<script>
	import { outerStore, innerStore } from './testStore.js'
	
	outerStore.subscribe(value => {
		if (value.inner === 'x') outerStore.set({ inner: 'y' })
	})
</script>

<p>
	$outerStore = {JSON.stringify($outerStore)}
</p>
<p>
	$innerStore = {JSON.stringify($innerStore)}
</p>

<button type="button" on:click={() => ($innerStore = 'a')}>Set inner to a</button>
<button type="button" on:click={() => ($innerStore = 'x')}>Set inner to x (should become y)</button>
```

Here, if we set `$innerStore = 'x'`, it will update `outerStore` to `{ inner: 'x' }` as expected. This will then cause the `subscribe` callback to be called, which immediately updates `outerStore` to `{ inner: 'y' }`.

**Expected behavior:** This external update to `outerStore` should trigger another update to `innerStore` so that it also becomes `'y'`.

**Actual behavior:** The external update is not propagated to `innerStore` - we now have a situation where `outerStore` is `{ inner: 'y' }` but `innerStore` is `'x'`!

See [REPL](https://svelte.dev/repl/682b35d98dd4484e9bfc11ac3b1f5f28?version=3.48.0).

**Comment by PixievoltNo1** [comment]
As a workaround, the subscriber-updater can instead be added to the inner store. Lines 4-6 of App.svelte would become:

```javascript
	innerStore.subscribe(value => {
		if (value === 'x') innerStore.set('y')
	})
```

I agree this should be fixed and will look into it.

**Comment by CherryDT** [comment]
Of course - the issue is just that in the real-world application, it's a piece of code in an entirely different place that does this subscriber-update, and it doesn't even know about any inner stores.

Thanks for looking into it!

**Comment by PixievoltNo1** [comment]
Fix released in v2.1.4. Thanks for the report!
